mantidproject/mantid
# Improve Performance of Loading Python Plugins During Startup

**Description of work.**
Replaces the searching of the file system for a list of python plugins with a CMake generated manifest of paths to those files. This will reduce the time spent accessing the file system during startup when launching mantid on slow drives. 

**To test:**
1. Launch workbench, ideally from a very slow or network drive. 
2. Check that all of the python algorithms have been loaded. 
3. The time taken to launch workbench should also have improved over a version without these changes. 

Fixes #28822 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
